### DISCUSSION TITLE
Could you share some metrics based on throughput?

### DISCUSSION THREAD

**Discussion by kharabela** [fluvio-community/fluvio#3890] [General]
Could you share some metrics based on throughput for fluvio?
![image](https://github.com/infinyon/fluvio/assets/2614068/0e81aa03-ad8d-481f-9cd5-c2761a384990)

**Comment by drc-infinyon** [comment]
Thanks for the question @kharabela

I creates a similar image comparing NATS, Kafka, and Fluvio. Going to use it to update our documentation.

![InfinyOn Comparison](https://github.com/infinyon/fluvio/assets/122817612/0ffc2c71-8c7a-46ab-85df-adea63872bf3)

In terms of Delivery Guarantees Fluvio supports at least once, at most once, and we support indempotence using deduplication which delivers the same output as exactly once - https://www.fluvio.io/docs/concepts/delivery-semantics/

Fluvio also supports ordering guarantees, persistence, replay-ability - https://www.fluvio.io/docs/concepts/data-consistency/

Throughput assumes a number of partitions and Fluvio can scale up to Kafka level throughputs while consuming a fraction of compute and memory.

On top of that the latest Stateful Services Development Kit enables stateful stream processing with offsets, states, and window based aggregations.

Hope this answers your question. Please let me know if you have any follow up.
